helix-term/src/application.rs

```diff
@@ -551,6 +551,21 @@ impl Application {
                                 }
                             };
 
+                        // Trigger a workspace/didChangeConfiguration notification after initialization.
+                        // This might not be required by the spec but Neovim does this as well, so it's
+                        // probably a good idea for compatibility.
+                        let config = self.editor.documents().find_map(|doc| {
+                            if doc.language_server().map(|server| server.id()) == Some(server_id) {
+                                doc.language_config()
+                                    .and_then(|config| config.config.clone())
+                            } else {
+                                None
+                            }
+                        });
```

Comment:
We register language servers by `language_config.scope`: https://github.com/helix-editor/helix/blob/5cfdb6dbca02e4558344feb01e2c9db7146dfec0/helix-lsp/src/lib.rs#L309-L314

I'd add a reverse lookup method that can scan over `(scope, server) in language_servers.iter()` `return scope where server.id() == id`

Reply:
I just saw that the LSP `Client` struct stores the config. I added a getter for it, that makes it a lot simpler.